File: WINGS_AI_Bootcamp/Elixir/FigurateNum.ex
import Bitwise  # Import bitwise operators like &&&

defmodule Figurate do
  def sq(n), do: n == :math.pow(:math.sqrt(n), 2) |> round

  def cube(n) do
    root = :math.pow(n, 1/3) |> round
    root * root * root == n
  end

  def power(n) when n > 0, do: (n &&& (n - 1)) == 0
  def power(_), do: false
end

IO.puts Figurate.sq(4)     # true
IO.puts Figurate.cube(27)  # true
IO.puts Figurate.power(4)  # true

File: WINGS_AI_Bootcamp/Python/FigurateNum.py
def sq(num: int) -> bool:
    return num == (num ** 0.5) ** 2
def cube(num: int) -> bool:
    c = round(num ** (1/3))
    return num == c ** 3
def power(num: int) -> bool:
    return num > 0 and (num & (num-1)) == 0

print(sq(4))
print(cube(27))
print(power(4))